### ISSUE TITLE
Bug: [triple-slash-reference] Reports errors when referencing `.d.ts` files

### ISSUE THREAD

**Issue by somebody1234** [typescript-eslint/typescript-eslint#7146]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&jsx=true&sourceType=module&showAST=es&code=PQgEB4CcFMDNpgOwMbVABwIYBcAWBeAIgDphNiATY7AZ0NGAD4g&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6fK4pW5eAQ2QALWtEQAzRKKZlE6KJOgB7aJHBgAviHVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
/// <reference path="./a.d.ts" />
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/triple-slash-reference": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
  }
}
```

### Expected Result

No error

### Actual Result

Error suggesting to switch to an `import` (in a CJS module too!)

### Additional Info

The playground repro is not a 1:1 repro as it is missing the `.d.ts`, and I don't believe there's a way to set the "code" file to `.js` instead of `.ts`

**Comment by bradzacher** [comment]
I'm not sure what the ask is here - the rule is there to ban triple-slash references - period.
You shouldn't be using `.d.ts` files in a codebase and should instead be declaring them as `.ts` and exporting the relevant types.

If you want triple slash references... then rule isn't for you!
